### PR TITLE
CA-393199: Disable external auth should clean pbis cache

### PR DESCRIPTION
**Pull request by liulinC**
The domain information is cached by PBIS. Such cache can help to accelerate query of domain information.

However, the cache data should be cleaned during disable external auth to make sure
- release cache as the domain is left
- rejoin domain refresh all the domain information

### CODE REVIEW

File: ocaml/xapi/extauth_plugin_ADpbis.ml
Comment (lindig) [inline ocaml/xapi/extauth_plugin_ADpbis.ml:1058]: The exception should be logged to help diagnose the problem.

**Review by psafont — APPROVED** [review]

**Review by Vincent-lau — APPROVED** [review]
Grammar issue, but I think this is confusing:

in the commit message you mean:
"release cache as the domain *has* left" right?

**Review by psafont — CHANGES_REQUESTED** [review]
Requesting changes since linding's comment is waiting

File: ocaml/xapi/extauth_plugin_ADpbis.ml
Comment (liulinC) [reply ocaml/xapi/extauth_plugin_ADpbis.ml:1058]: > The exception should be logged to help diagnose the problem.

I ignore the exception by intention as we do not really care about the error, (and ignored)
this is best to have clean.
But I have updated the code anyway as it always better to log errors.

**Comment by liulinC** [comment]
Sorry, I accidentally create a new branch in this repo, (instead of my private repo). 
Will delete the branch after this merge.

**Review by lindig — APPROVED** [review]

File: ocaml/xapi/extauth_plugin_ADpbis.ml
Comment (lindig) [inline ocaml/xapi/extauth_plugin_ADpbis.ml:1059]: I believe this the wrong code and suggest to use `Printexc.to_string`

**Comment by lindig** [comment]
This was not tested

**Comment by liulinC** [comment]
> tested

The original code was test in my local env, 
I did not re-test `Printexc.to_string` as it is low risk.
Thanks compiler detected the grammar issue.

**Comment by robhoes** [comment]
I believe @psafont 's request has been addressed, so I'll dismiss his review and merge the PR.

**Pull request merged**
